assets/js/app.js

```diff
@@ -0,0 +1,53 @@
+let productsArray = [];  
+
+function collectData() {
+    const productName = document.getElementById('productName').value;
+    const price = document.getElementById('price').value;
+    const select = document.getElementById('select').value;
+    const URL = document.getElementById('URL').value;
+    return { productName, price, select, URL };
+}
+
+function generateHTML(data, index) {
+    const newHTML = `
+    <tr class="table-warning" data-index="${index}">
+        <td>${data.productName}</td>
+        <td>${data.price}</td>
+        <td>${data.select}</td>
+        <td><img class="size" src="${data.URL}" /></td>
+        <td><button class="btn btn-danger" onclick="removeProduct(${index})">Remove</button></td>
+    </tr>`;
+    return newHTML;
+}
+
+function renderHTML() {
+    const productsContainer = document.getElementById('productsContainer');
+    productsContainer.innerHTML = '';  
+    productsArray.forEach((product, index) => {
+        const newHTML = generateHTML(product, index);
+        productsContainer.innerHTML += newHTML;
```

Comment:
bad practice to update the DOM in a loop. very not efficient. the browser will re-render with every loop iteration. better cumulate the result in a string and then after the loop make a single assignment to invoke re-rendering